R/endpoint.R

```diff
@@ -164,5 +167,22 @@ pkgapi_endpoint <- R6::R6Class(
         args <- self$inputs$validate(given)
         do.call(self$run, args)
       }, error = pkgapi_process_error)
+    },
+
+    ##' @description Create a plumber endpoint
+    ##'
+    ##' @param envir Environment as used by plumber (currently unclear)
+    ##'
+    ##' @param validate Logical, allowing override of validation at the api
+    ##'   level.  This takes precedence over the value set when creating the
+    ##'   endpoint.
+    create = function(envir, validate) {
+      if (!identical(validate, self$validate)) {
+        self <- self$clone()
+        self$validate <- validate
+      }
```

Comment:
This also looks kind of strange to me. Does this do anything different than `self$validate <- validate`?

Reply:
Because endpoints have reference semantics, without the clone we would change the incoming endpoint object (see https://github.com/reside-ic/pkgapi/pull/7/files#diff-417e3735f0563c5835f4f8fd516c1d5cR136 which verifies that this does not happen).  This means that one could create all endpoints with `validate = TRUE` and construct the api with `validate = FALSE` and each endpoint would validate when called separately but not when called from the API - this may be a useful pattern for simple packages that do not need to create endpoints in functions

Reply:
Nice - that makes sense too. Thanks